### PR TITLE
Make package testable

### PR DESCRIPTION
**Pull request by szepeviktor**
Please see comments.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     }
   ],
   "require": {
-    "php": "8.0.*|8.1.*",
+    "php": "~8.0 || ~8.1",
```
Comment (szepeviktor) [inline composer.json:16]: This is the canonical form.

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,9 @@
   },
   "require-dev": {
     "phpstan/phpstan": "^1.6",
-    "phpunit/phpunit": ">=8.0"
+    "phpunit/phpunit": ">=8.0",
+    "psr/http-factory": "^1.0",
+    "symfony/http-client": "^6.0"
```
Comment (szepeviktor) [inline composer.json:27]: Without `symfony/http-client` `composer update` simply fails.

File: composer.json [hunk 3]
```diff
@@ -22,7 +22,9 @@
   },
   "require-dev": {
     "phpstan/phpstan": "^1.6",
-    "phpunit/phpunit": ">=8.0"
+    "phpunit/phpunit": ">=8.0",
+    "psr/http-factory": "^1.0",
```
Comment (szepeviktor) [inline composer.json:26]: `RequestFactoryInterface` is in this package.

**Comment by lostfocus** [comment]
Thanks! I'll have a look at it later this week when I find the time.
I was about to deprecate the library, though - I started to move a bit of the work to https://github.com/php-weather/ (before life and other distractions got in the way)
The plan is to build up a good library there.

**Pull request closed without merging**